<launch>
    <arg name="launch_prefix" default="" /> <!-- set to value="gdbserver localhost:10000" for remote debugging -->
    <arg name="node_namespace" default="apriltag_ros_continuous_node" />
    <arg name="camera_frame" default="camera_link" />
    <arg name="image_topic" default="image" />

    <!-- Set parameters -->
    <rosparam command="load" file="$(find apriltag_detection)/config/settings.yaml" ns="$(arg node_namespace)" />
    <rosparam command="load" file="$(find apriltag_detection)/config/tags.yaml" ns="$(arg node_namespace)" />

    <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)" >
        <!-- Remap topics from those used in code to those on the ROS network -->
        <remap from="image_rect" to="sync_image" />
        <remap from="camera_info" to="sync_camera_info" />

        <param name="camera_frame" type="str" value="camera_link" />
        <param name="publish_tag_detections_image" type="bool" value="true" />      <!-- default: false -->
    </node>

    <node pkg="apriltag_detection" type="simple_sync.py" name="image_sync" output="screen"/>

    <node pkg="apriltag_detection" type="apriltag_detection.py" name="apriltag_detection" output="screen"/>

    <node pkg="tf" type="static_transform_publisher" name="camera_to_base" args="0.03 0 0.1 0 0 0 base_link camera_link 100"/>
</launch>